fs/btrfs/zlib.c

```diff
@@ -36,7 +36,7 @@ static void zlib_free_workspace(struct list_head *ws)
 	kfree(workspace);
 }
 
-static struct list_head *zlib_alloc_workspace(void)
+static struct list_head *zlib_alloc_workspace(unsigned level)
```

Comment:
We should call `zlib_set_level(ws, level)` before we return the workspace, since if we allocate a workspace we don't call `set_level()`.